Simperium-OSX/SPAuthenticationTextField.m

```diff
@@ -72,34 +71,44 @@ - (void)dealloc {
 - (instancetype)initWithFrame:(NSRect)frame secure:(BOOL)secure {
     self = [super initWithFrame:frame];
     if (self) {
-        // Center the textField vertically
-        int paddingX = 10;
-        int fontSize = 20;
-        CGFloat fieldHeight = [[SPAuthenticationConfiguration sharedInstance] regularFontHeightForSize:fontSize];
-        CGFloat fieldY = (self.frame.size.height - fieldHeight) / 2;
-        CGRect textFrame = NSMakeRect(paddingX, fieldY, frame.size.width-paddingX*2, fieldHeight);
-
-        Class textFieldClass = secure ? [SPSecureTextField class] : [SPTextField class];
-        _textField = [[textFieldClass alloc] initWithFrame:textFrame];
-        NSFont *font = [NSFont fontWithName:[SPAuthenticationConfiguration sharedInstance].regularFontName size:fontSize];
-        [_textField setFont:font];
-        [_textField setTextColor:[NSColor colorWithCalibratedWhite:0.1 alpha:1.0]];
-        [_textField setDrawsBackground:NO];
-        [_textField setBezeled:NO];
-        [_textField setBordered:NO];
-        [_textField setFocusRingType:NSFocusRingTypeNone];
-        [[_textField cell] setWraps:NO];
-        [[_textField cell] setScrollable:YES];
-        [self addSubview:_textField];
-        
-        NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
-        [nc addObserver:self selector:@selector(handleTextFieldDidBeginEditing:) name:SPTextFieldDidBecomeFirstResponder object:_textField];
-        [nc addObserver:self selector:@selector(handleTextFieldDidFinishEditing:) name:NSControlTextDidEndEditingNotification object:_textField];
+        [self setupInterface:secure];
     }
-    
+
     return self;
 }
 
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    [self setupInterface:self.secure];
```

Comment:
When this view is initialised from the NIB file, is it possible (or do we need?) to set `secure` flag from the NIB? From what I see it will always be `false` 🤔

Reply:
Okay, so this flag can be set via IB :-)